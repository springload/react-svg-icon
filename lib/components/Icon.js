import React from 'react';

export default React.createClass({

  displayName: 'Icon',

  propTypes: {
    name: React.PropTypes.string.isRequired,
    className: React.PropTypes.string
  },

  render() {
    const className = ['i'];
    if (this.props.className) {
      className.push(this.props.className);
    }

    return (
      <svg className={className.join(' ')}>
        <use xlinkHref={'#i-' + this.props.name} />
      </svg>
    );
  }

});
