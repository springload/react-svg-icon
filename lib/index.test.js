import React from 'react';
import { shallow } from 'enzyme';
import Icon from './index';

describe('Icon', () => {
    it('renders', () => {
        expect(shallow(<Icon name="rocket" />)).toMatchSnapshot();
    });

    it('#className', () => {
        expect(shallow(<Icon name="rocket" className="i--red" />)).toMatchSnapshot();
    });

    it('#title', () => {
        expect(shallow(<Icon name="rocket" title="Test title" />)).toMatchSnapshot();
    });
});
