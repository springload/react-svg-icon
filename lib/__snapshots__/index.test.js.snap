// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon #className 1`] = `
<svg
  aria-hidden={true}
  className="i i--red"
>
  <use
    aria-hidden={null}
    xlinkHref="#i-rocket"
  />
</svg>
`;

exports[`Icon #title 1`] = `
<svg
  aria-hidden={null}
  className="i "
>
  <title>
    Test title
  </title>
  <use
    aria-hidden={true}
    xlinkHref="#i-rocket"
  />
</svg>
`;

exports[`Icon renders 1`] = `
<svg
  aria-hidden={true}
  className="i "
>
  <use
    aria-hidden={null}
    xlinkHref="#i-rocket"
  />
</svg>
`;
